CMakeLists.txt

```diff
@@ -0,0 +1,187 @@
+cmake_minimum_required(VERSION 3.26)
+
+set(VCPKG_USE_HOST_TOOLS ON)
+
+project(dosbox-staging
+		LANGUAGES C CXX
+		VERSION 0.82.0
+)
+
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+# TODO: Enable certain warnings-as-errors for at least MSVC, Clang & GCC
+
+option(OPT_DEBUG "Enable debugging" $<IF:$<CONFIG:Debug>,ON,OFF>)
+option(OPT_HEAVY_DEBUG "Enable heavy debugging" OFF)
+
+if (OPT_HEAVY_DEBUG)
+	set(OPT_DEBUG ON CACHE INTERNAL "")
+endif()
+
+include(CheckIncludeFile)
+include(CheckIncludeFiles)
+include(CheckSymbolExists)
+
+if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
+	set(DOSBOX_PLATFORM_WINDOWS ON)
+elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+	set(DOSBOX_PLATFORM_MACOS ON)
+elseif(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+	set(DOSBOX_PLATFORM_LINUX ON)
+else()
+	message(FATAL_ERROR "Unknown system ${CMAKE_SYSTEM_NAME}")
+endif()
+
+# TODO: Proper compiler simulation detection (i.e. clang on MSVC)
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+	set(DOSBOX_COMPILER_CLANG ON)
+elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+	set(DOSBOX_COMPILER_GCC ON)
+elseif (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" AND CMAKE_CXX_SIMULATE_ID STREQUAL "" AND CMAKE_C_SIMULATE_ID STREQUAL "")
+	set(DOSBOX_COMPILER_MSVC ON)
+else()
+	message(NOTICE "Unknown compiler")
+endif()
+
+
+find_package(Git)
+if(Git_FOUND)
+	execute_process(
+			COMMAND ${GIT_EXECUTABLE} log -1 --format=%h
+			WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
+			OUTPUT_VARIABLE VCS_TAG
+			OUTPUT_STRIP_TRAILING_WHITESPACE
+	)
+else()
+	set(VCS_TAG "dirty")
+endif()
+
+check_include_file("unistd.h" HAS_UNISTD)
+if (NOT HAS_UNISTD)
+	configure_file(src/platform/visualc/unistd.h ${CMAKE_CURRENT_BINARY_DIR}/include/unistd.h @ONLY)
+endif()
+
+### config.h shenanigans
+
+file(READ src/config.h.in CONFIG_CONTENTS)
+string(REPLACE "#mesondefine" "#cmakedefine" CONFIG_CONTENTS "${CONFIG_CONTENTS}")
+string(REPLACE "#cmakedefine C_TARGETCPU" "#cmakedefine C_TARGETCPU @C_TARGETCPU@" CONFIG_CONTENTS "${CONFIG_CONTENTS}")
+string(REPLACE "#cmakedefine CUSTOM_DATADIR" "#cmakedefine CUSTOM_DATADIR \"@CUSTOM_DATADIR@\"" CONFIG_CONTENTS "${CONFIG_CONTENTS}")
+
+set_property(GLOBAL PROPERTY FILE_CONTENTS "${CONFIG_CONTENTS}")
+
+function(setConfig01 VAR VALUE)
+	get_property(fc GLOBAL PROPERTY FILE_CONTENTS)
+	string(REPLACE "#cmakedefine ${VAR}" "#cmakedefine01 ${VAR}" fc "${fc}")
+	set_property(GLOBAL PROPERTY FILE_CONTENTS "${fc}")
+	set(${VAR} ${VALUE} PARENT_SCOPE)
+endfunction()
+
+if (CMAKE_SYSTEM_PROCESSOR STREQUAL "AMD64")
```

Comment:
My pretty standard Linux build failed here:

```
CMake Error at CMakeLists.txt:84 (message):
  Unknown processor x86_64
```

Reply:
Of course, if you read the whole PR you'll notice I said this will *only* work for Windows right now.

Reply:
@dreamer We agreed with @Arcnor to merge in his work in stages in many small PRs, versus one big PR when everything is working. So yeah, this is the initial Windows only PR.